### PR TITLE
[cleanup] Misc. (for 2023-11 release)

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (bashonly, coletdjnz, dirkf, gamer191, seproDev)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
:sparkles::warning::mute:

Add support for the requests library as an alternative to urllib3, and update the YoutubePlaylistIE extractor and the changelog accordingly. Fix a logging issue with the requests library in `yt_dlp/networking/_requests.py`.

> _`requests` library_
> _new feature for youtube-dl_
> _autumn changelog cut_

### Walkthrough
*  Add support for the requests HTTP library, which enables HTTPS proxies and persistent connections ([link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-323808d1d1a45be3d06ea051c76192878238cd5c25e74087fbdad00eab2c6a16R101-R106), [link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-ef5ef94777b4b7a8fd1579b2053264fb5068b9e7baf33e95c23d3541a9eab1e4L258-R259), [link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-b7b9f6790de4427214b61939432e667d95b929d07fd918b9da1a36d7996cc506L6690-R6690))
   * Override the changelog entry for this feature in `changelog_override.json` ([link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-323808d1d1a45be3d06ea051c76192878238cd5c25e74087fbdad00eab2c6a16R101-R106))
   * Implement a wrapper class `RequestsRH` that adapts the requests library to the request handler interface ([link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-ef5ef94777b4b7a8fd1579b2053264fb5068b9e7baf33e95c23d3541a9eab1e4L258-R259))
      * Suppress a warning message from the urllib3 logger by lowering its level to ERROR ([link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-ef5ef94777b4b7a8fd1579b2053264fb5068b9e7baf33e95c23d3541a9eab1e4L258-R259))
   * Update the test case for `YoutubePlaylistIE` to expect two more warnings related to unavailable videos ([link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-b7b9f6790de4427214b61939432e667d95b929d07fd918b9da1a36d7996cc506L6690-R6690))
      * Reflect the new behavior of the extractor, which uses the requests library and handles unavailable videos differently ([link](https://github.com/yt-dlp/yt-dlp/pull/8510/files?diff=unified&w=0#diff-b7b9f6790de4427214b61939432e667d95b929d07fd918b9da1a36d7996cc506L6690-R6690))

</details>

### CODE REVIEW

File: yt_dlp/networking/_requests.py [hunk 1]
```diff
@@ -255,7 +255,8 @@ def __init__(self, *args, **kwargs):
         handler.setFormatter(logging.Formatter('requests: %(message)s'))
         handler.addFilter(Urllib3LoggingFilter())
         logger.addHandler(handler)
-        logger.setLevel(logging.WARNING)
+        # Use ERROR to suppress pool reuse warning (could filter instead?)
+        logger.setLevel(logging.ERROR)
```
Comment (coletdjnz) [inline yt_dlp/networking/_requests.py:259]:
```suggestion
        logger.setLevel(logging.ERROR)
```

don't think this needs a comment imo. was supposed to be like this from the start

Comment (Grub4K) [reply yt_dlp/networking/_requests.py:259]: I added a comment since we might want to explore the filter option instead of setting to `ERROR`

Comment (Grub4K) [reply yt_dlp/networking/_requests.py:259]: I'll make it be a TODO to use a filter instead then

File: yt_dlp/extractor/cbc.py [hunk 2]
```diff
@@ -387,7 +388,7 @@ def _find_secret_formats(self, formats, video_id):
         url = re.sub(r'(Manifest\(.*?),format=[\w-]+(.*?\))', r'\1\2', base_url)
 
         secret_xml = self._download_xml(url, video_id, note='Downloading secret XML', fatal=False)
-        if not secret_xml:
+        if not isinstance(secret_xml, xml.etree.ElementTree.Element):
```
Comment (pukkandan) [inline yt_dlp/extractor/cbc.py:391]: this and related changes should be a separate commit imo, not squashed into cleanup
